### ISSUE TITLE
Question: History directory usage.

### ISSUE THREAD

**Issue by tfabris** [danielpalme/ReportGenerator#314]
I'm considering using ReportGenerator, but I'm trying to figure out if it will work correctly in my CI system before I dive in too far. My questions are about the history directory option:

     History directory:  Optional directory for storing persistent coverage information.
                         Can be used in future reports to show coverage evolution.

This seems like a killer feature, and I'm very interested in it. I have questions:

- Is the history directory used only when each report is being generated? Or does the directory also get used at the time when the report is viewed? My intention is for the report to be viewed from a server which is different than the server on which the report is generated. The server where the final HTML report will be hosted and viewed might not be able to get permissions to reach the history directory on the other server.

- Are there options to limit the size of the history? For instance, keeping only a year, or a month, of history, for example. If not, I wonder if the files are obvious enough so that I can write a simple script to keep the history from eventually filling the hard disk.

I haven't had a chance to successfully try the tool yet. My reports are currently coming up empty, so I'm fiddling around with the options until I can get something working.

Thanks!

**Comment by tfabris** [comment]
Answering part of my own question: Even though I'm not getting a successful report yet, I can at least experiment with the history folder a bit. I see this as part of the verbose log output:

     "MaximumNumberOfHistoricCoverageFiles": 100

That's excellent. Can that be specified on the command line somewhere, if I want longer or shorter history? I tried putting -MaximumNumberOfHistoricCoverageFiles:256 directly on the command line but that didn't work. In any case, if I want to shorten the files, the filenames are very clear, so that's not a problem. But I might want a longer history than 100, though.

My other question about file access still stands, though.

Thanks!

**Comment by danielpalme** [comment]
> * Is the history directory used only when each report is being generated? Or does the directory also get used at the time when the report is viewed? My intention is for the report to be viewed from a server which is different than the server on which the report is generated. The server where the final HTML report will be hosted and viewed might not be able to get permissions to reach the history directory on the other server.

The history directory is only used when the report is being generated. The resulting HTML files contain everything and no access to that directory is required.

> * Are there options to limit the size of the history? For instance, keeping only a year, or a month, of history, for example. If not, I wonder if the files are obvious enough so that I can write a simple script to keep the history from eventually filling the hard disk.

Please use the following comand line parameter: `settings:maximumNumberOfHistoricCoverageFiles=100`  
See also: https://github.com/danielpalme/ReportGenerator/wiki/Settings#other-settings

> I haven't had a chance to successfully try the tool yet. My reports are currently coming up empty, so I'm fiddling around with the options until I can get something working.

You may want to have a look at the [configuration tool](https://danielpalme.github.io/ReportGenerator/usage.html) to get the command line parameters right.  
Let me know if you need further help. If that's the case, the log output could be helpful.

**Comment by tfabris** [comment]
Thank you so much for your excellent answers. I will use the "settings" command line parameter as you suggested.

The reason that ReportGenerator was unsuccessful for me earlier, was that I was generating the XML file from DotCover with the wrong XML output type to begin with. In order for it to work, I had to use "/ReportType=DetailedXML" as the parameter when running DotCover to generate the file. This was already made clear in your documentation, and I had suspected that it might be the issue, so after changing that parameter in my CI system, now ReportGenerator is cranking along and building its first full report as I type this.

Thank you so much!
